hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/util/WrappingProcessorMetaSupplier.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.impl.util;
+
+import com.hazelcast.jet.Processor;
+import com.hazelcast.jet.ProcessorMetaSupplier;
+import com.hazelcast.jet.ProcessorSupplier;
+import com.hazelcast.nio.Address;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * A {@link ProcessorMetaSupplier} which wraps another {@code
+ * ProcessorMetaSupplier} with one that will wrap its processors using
+ * {@code wrapperSupplier}.
+ */
+public final class WrappingProcessorMetaSupplier implements ProcessorMetaSupplier {
+    private ProcessorMetaSupplier wrapped;
+    private Function<Processor, Processor> wrapperSupplier;
+
+    public WrappingProcessorMetaSupplier(ProcessorMetaSupplier wrapped, Function<Processor,
+            Processor> wrapperSupplier) {
```

Comment:
We shouldn't have a line break between two type parameters. The whole second argument declaration should be on the next line.